pkg/reftracker/ref_tracker.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package reftracker
+
+import (
+	"fmt"
+	"strings"
+)
+
+var appExists struct{}
+
+type AppRefTracker struct {
```

Comment:
we got maps that are accessed in concurrent so need locks.

Reply:
Added lock